src/lib/support/CHIPArgParser.cpp

```diff
@@ -332,48 +394,115 @@ bool ParseArgs(const char * progName, int argc, char * const argv[], OptionSet *
         goto done;
     }
 
+#ifdef CONFIG_NON_POSIX_GETOPT_LONG
```

Comment:
use `#if` instead of `#ifdef` to force defines to exist as need.

This does mean that th define will have to be set somewhere else than just a specific file.